util/buffer.cc

```diff
@@ -0,0 +1,45 @@
+#include "buffer.h"
+#include "logger.h"
+#include "gzip.h"
+
+namespace atlas {
+namespace util {
+
+void Buffer::append(void* data, size_t data_size) {
+  memory = static_cast<char*>(realloc(memory, size + data_size + 1));
```

Comment:
This potentially results in copying the existing buffer for each append if the existing area cannot be expanded in-place. Wouldn't it be better to have some excess capacity in the buffer to limit the worst case number of copies?